app/src/main/java/com/github/sdpteam15/polyevents/view/activity/admin/EventManagementActivity.kt

```diff
@@ -1,13 +1,356 @@
+            if (isChecked) {
+                nbpart.visibility = View.VISIBLE
+                nbpart.setText(defaultPartNb)
+            } else {
+                nbpart.visibility = View.INVISIBLE
+                nbpart.setText(emptyPartNb)
+            }
+        }
+    }
+
+    private fun setButtonListener() {
+        findViewById<Button>(R.id.btnStartDate).setOnClickListener {
+            dialogStartDate.show()
+        }
+        findViewById<Button>(R.id.btnEndDate).setOnClickListener {
+            dialogEndDate.show()
+        }
+        this.findViewById<Button>(R.id.btnStartTime).setOnClickListener {
+            dialogStartTime.show()
+        }
+        findViewById<Button>(R.id.btnEndTime).setOnClickListener {
+            dialogEndTime.show()
+        }
+    }
+
+    @SuppressLint("UseSwitchCompatOrMaterialCode")
+    private fun setupViewInActivity(onCallback: Boolean) {
+        val btnManage = findViewById<Button>(R.id.btnManageEvent)
+        val switch = findViewById<Switch>(R.id.swtLimitedEvent)
+        val descET = findViewById<EditText>(R.id.eventManagementDescriptionField)
+        val nameET = findViewById<EditText>(R.id.eventManagementNameField)
+        val nbpart = findViewById<EditText>(R.id.etNbPart)
+
+        if (!onCallback) {
+            btnManage.text = getString(R.string.create_event_btn_text)
+            nameET.setText("")
+            descET.setText("")
+            switch.isChecked = false
+            nbpart.visibility = View.INVISIBLE
+            nbpart.setText(emptyPartNb)
+        } else {
+            val event = observableEvent.value!!
+            btnManage.text = getString(R.string.update_event_btn_text)
+            nameET.setText(event.eventName)
+            descET.setText(event.description)
+
+            nbpart.setText(emptyPartNb)
+            nbpart.visibility = View.INVISIBLE
+
+            if (event.isLimitedEvent()) {
+                nbpart.setText(event.getMaxNumberOfSlots()!!.toString())
+                nbpart.visibility = View.VISIBLE
+            }
+            switch.isChecked = event.isLimitedEvent()
+            dateStart.postValue(event.startTime!!)
+            dateEnd.postValue(event.endTime!!)
+            updateTextDate(typeEnd)
+            updateTextDate(typeStart)
+        }
+    }
+
+    private fun manageButtonSetup() {
+        val btnManage = findViewById<Button>(R.id.btnManageEvent)
+        if (isCreation) {
+            btnManage.setOnClickListener {
+                if (verifyCondition()) {
+                    currentDatabase.eventDatabase!!.createEvent(getInformation()).observe(this) {
+                        redirectOrDisplayError(
+                            getString(R.string.event_creation_success),
+                            getString(R.string.event_creation_failed),
+                            it.value
+                        )
+                    }
+                }
```

Comment:
else showToast("Unable to create Event, please check the dates you entered are valid") or something like this